### ISSUE TITLE
Identification problem

### ISSUE THREAD

**Issue by marlonwq** [Random-user420/bestKabu#20]
When you are on the website https://digikabu.de/ and click on the extension it displays the standard redirect message to the website indicating that you need to be on it for the extension to work, it doesn't make sense because you are already on the website .

Perhaps this could be resolved with a verification/identification functionality.

https://github.com/user-attachments/assets/b1383515-d9db-4613-a954-8d1f8c42be63

**Comment by Random-user420** [comment]
Where you on the website and then enabled the extension? Because then the extension isn't invoked by the browser on the website, so I can't fix that. If not then provide (if you can) more information and I'll test some things, because that's weird. (edit) And if it's the first case: we can't actually detect the current website, so the page would only be refreshed and not opened in a another tab.

**Comment by marlonwq** [comment]
> Where you on the website and then enabled the extension? Because then the extension isn't invoked by the browser on the website, so I can't fix that. If not then provide (if you can) more information and I'll test some things, because that's weird. (edit) And if it's the first case: we can't actually detect the current website, so the page would only be refreshed and not opened in a another tab.

When I am using other browsers like Chorme, the extension works normally, there is no need to click on it and then click on the website redirect link.

When I use opera it doesn't work the way it "should". It's as if she wasn't identifying that I'm already on the site to start working...

Then what I showed in the video happens.

**Comment by Random-user420** [comment]
I just tested it on Linux with the opera flatpak ant it worked just fine. It could be an issue with some other version of opera, but the behavior in the attached video of mine, is just as expected. Maybe update opera or switch to a dev version. But since I can't test on windows, I'll let a other contributior of the project know that they should take a look into it. And thank you for the report.

A thing that I just researched: Opera may be doing some weird things with permissions, so check that it can access digikabu.de. And The opera built in privacy setting are maybe influencing the functionality of the extension. 

https://github.com/user-attachments/assets/31457409-48ca-4e2d-8b6a-1cbb11d5aed0
